cpp/ql/test/library-tests/dataflow/DefaultTaintTracking/test_default_diff.ql

```diff
@@ -0,0 +1,31 @@
+import cpp
+import semmle.code.cpp.security.Security
+import semmle.code.cpp.security.TaintTracking as ASTTaintTracking
+import semmle.code.cpp.ir.dataflow.DefaultTaintTracking as IRDefaultTaintTracking
+
+predicate astFlow(Location sourceLocation, Location sinkLocation) {
+  exists(Expr source, Element sink |
+    ASTTaintTracking::tainted(source, sink) and
+    sourceLocation = source.getLocation() and
+    sinkLocation = sink.getLocation()
+  )
+}
+
+predicate irFlow(Location sourceLocation, Location sinkLocation) {
+  exists(Expr source, Element sink |
+    IRDefaultTaintTracking::tainted(source, sink) and
+    sourceLocation = source.getLocation() and
+    sinkLocation = sink.getLocation()
+  )
+}
+
+from Location source, Location sink, string note
```

Comment:
Why compare the sources and sinks by `Location` rather than comparing them as `Expr` and `Element` respectively? If the two libraries find elements that are different, then I think we want to know about it even if they have the same location.